torch_xla/csrc/ops/dynamic_ir.cpp

```diff
@@ -35,13 +35,18 @@ SizeNode::SizeNode(torch::lazy::Value input, size_t dim)
 };
 
 int64_t SizeNode::getDynamicValue() const {
+  if (dyanmic_value_computed_) {
+    XLA_COUNTER("CachedSizeNodeValue", 1);
+    return runtime_size_;
+  }
   torch::lazy::NodePtr cloned =
       torch::lazy::MakeNode<SizeNode>(operands_[0], dim_);
   std::vector<XLATensorPtr> dummy_size_tensors = {
       XLATensor::Create(cloned, *GetDefaultDevice(), at::ScalarType::Long)};
-  // TODO: cache the result
   std::vector<at::Tensor> res = XLATensor::GetTensors(&dummy_size_tensors);
-  return res[0].sum().item().toInt();
+  runtime_size_ = res[0].sum().item().toInt();
```

Comment:
Might be irrelevant to this pr, I'm wondering what it means by ".sum()". It seems sum is defined in TensorBody.h as
```
inline at::Tensor Tensor::sum(c10::optional<at::ScalarType> dtype) const {
    return at::_ops::sum::call(const_cast<Tensor&>(*this), dtype);
}
```
and at::_ops::sum is defined in sum_ops.h. But I couldn't find any comment so it is not clear to me what it means.

Reply:
`sum` just sum the tensor, I get since this tensor is always a scalar, I don't need `sum` here.